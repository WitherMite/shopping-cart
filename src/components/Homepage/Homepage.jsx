import { Link } from "react-router-dom";
import heroImage from "../../assets/hero-image.png";

export default function Homepage() {
  return (
    <>
      <header className="homepage-hero">
        <div className="hero-image">
          <img src={heroImage} alt="" />
          <a href="https://www.freepik.com/free-photo/beauty-fashion-concept-full-length-young-woman-sticking-head-inside-purse-searching-somet_40245224.htm#fromView=search&page=1&position=49&uuid=782d1c21-379a-4328-b57e-dd88b5fb1745&new_detail=true">
            Image by benzoix on Freepik
          </a>
        </div>
        <h1>Welcome to my store!</h1>
        <p>
          Lorem ipsum dolor, sit amet consectetur adipisicing elit. Unde cumque
          porro minima debitis a quis aliquid quidem. Ipsa voluptas tenetur
          earum? Et a rem, nulla adipisci totam similique tempore. Sequi fuga
          quidem veniam quia voluptate! Maiores magni dicta deserunt et rem
          perferendis, eaque recusandae culpa obcaecati id quam, velit labore.
          Accusantium. Lorem ipsum dolor sit amet, consectetur adipisicing elit.
          Aliquid enim dolorum impedit consequuntur deleniti pariatur harum
          officiis molestias doloribus nam dolorem, similique quam minus maxime
          fugit magnam fuga accusantium hic? Lorem ipsum dolor sit amet
          consectetur adipisicing elit. Libero sequi iure omnis asperiores.
          Reiciendis ut doloribus inventore libero, corporis quo hic quisquam
          similique nostrum nam saepe. Perferendis sequi libero maxime!
        </p>
      </header>
      <section className="homepage-content">
        <h2>Shop our selection:</h2>
        <div className="categories">
          <div className="category-card">
            <img
              src="https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_.jpg"
              alt="A men's T-shirt"
            />
            <h3>Men&apos;s wear</h3>
            <Link to="/store">
              <button className="shop-btn">Shop</button>
            </Link>
          </div>
          <div className="category-card">
            <img
              src="https://fakestoreapi.com/img/61sbMiUnoGL._AC_UL640_QL65_ML3_.jpg"
              alt="A ring encrusted with diamonds"
            />
            <h3>Jewelry</h3>
            <Link to="/store">
              <button className="shop-btn">Shop</button>
            </Link>
          </div>
          <div className="category-card">
            <img
              src="https://fakestoreapi.com/img/51eg55uWmdL._AC_UX679_.jpg"
              alt="A women's short sleeve shirt"
            />
            <h3>Women&apos;s wear</h3>
            <Link to="/store">
              <button className="shop-btn">Shop</button>
            </Link>
          </div>
          <div className="category-card">
            <img
              src="https://fakestoreapi.com/img/81Zt42ioCgL._AC_SX679_.jpg"
              alt="A curved ultrawide computer monitor"
            />
            <h3>Technology</h3>
            <Link to="/store">
              <button className="shop-btn">Shop</button>
            </Link>
          </div>
        </div>
      </section>
    </>
  );
}
