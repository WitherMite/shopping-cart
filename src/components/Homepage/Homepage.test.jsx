import { createMemoryRouter, RouterProvider } from "react-router-dom";
import { render } from "@testing-library/react";
import { it, expect } from "vitest";
import Homepage from "./Homepage.jsx";

it("Renders static homepage", () => {
  const router = createMemoryRouter([{ path: "/", element: <Homepage /> }]);
  const { container } = render(<RouterProvider router={router} />);
  expect(container).toMatchSnapshot();
});
