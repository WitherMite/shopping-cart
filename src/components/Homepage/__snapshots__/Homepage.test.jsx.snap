// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders static homepage 1`] = `
<div>
  <header
    class="homepage-hero"
  >
    <div
      class="hero-image"
    >
      <img
        alt=""
        src="/src/assets/hero-image.png"
      />
      <a
        href="https://www.freepik.com/free-photo/beauty-fashion-concept-full-length-young-woman-sticking-head-inside-purse-searching-somet_40245224.htm#fromView=search&page=1&position=49&uuid=782d1c21-379a-4328-b57e-dd88b5fb1745&new_detail=true"
      >
        Image by benzoix on Freepik
      </a>
    </div>
    <h1>
      Welcome to my store!
    </h1>
    <p>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Unde cumque porro minima debitis a quis aliquid quidem. Ipsa voluptas tenetur earum? Et a rem, nulla adipisci totam similique tempore. Sequi fuga quidem veniam quia voluptate! Maiores magni dicta deserunt et rem perferendis, eaque recusandae culpa obcaecati id quam, velit labore. Accusantium. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquid enim dolorum impedit consequuntur deleniti pariatur harum officiis molestias doloribus nam dolorem, similique quam minus maxime fugit magnam fuga accusantium hic? Lorem ipsum dolor sit amet consectetur adipisicing elit. Libero sequi iure omnis asperiores. Reiciendis ut doloribus inventore libero, corporis quo hic quisquam similique nostrum nam saepe. Perferendis sequi libero maxime!
    </p>
  </header>
  <section
    class="homepage-content"
  >
    <h2>
      Shop our selection:
    </h2>
    <div
      class="categories"
    >
      <div
        class="category-card"
      >
        <img
          alt="A men's T-shirt"
          src="https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_.jpg"
        />
        <h3>
          Men's wear
        </h3>
        <a
          href="/store"
        >
          <button
            class="shop-btn"
          >
            Shop
          </button>
        </a>
      </div>
      <div
        class="category-card"
      >
        <img
          alt="A ring encrusted with diamonds"
          src="https://fakestoreapi.com/img/61sbMiUnoGL._AC_UL640_QL65_ML3_.jpg"
        />
        <h3>
          Jewelry
        </h3>
        <a
          href="/store"
        >
          <button
            class="shop-btn"
          >
            Shop
          </button>
        </a>
      </div>
      <div
        class="category-card"
      >
        <img
          alt="A women's short sleeve shirt"
          src="https://fakestoreapi.com/img/51eg55uWmdL._AC_UX679_.jpg"
        />
        <h3>
          Women's wear
        </h3>
        <a
          href="/store"
        >
          <button
            class="shop-btn"
          >
            Shop
          </button>
        </a>
      </div>
      <div
        class="category-card"
      >
        <img
          alt="A curved ultrawide computer monitor"
          src="https://fakestoreapi.com/img/81Zt42ioCgL._AC_SX679_.jpg"
        />
        <h3>
          Technology
        </h3>
        <a
          href="/store"
        >
          <button
            class="shop-btn"
          >
            Shop
          </button>
        </a>
      </div>
    </div>
  </section>
</div>
`;
